src/Apps/Sell/SubmissionContext.tsx

```diff
@@ -0,0 +1,162 @@
+import * as Yup from "yup"
+import { SubmissionContextUpdateSubmissionMutation } from "__generated__/SubmissionContextUpdateSubmissionMutation.graphql"
+import { createContext, useState } from "react";
+import { graphql } from "react-relay";
+import { isEqual } from "lodash";
+import { useMutation } from "Utils/Hooks/useMutation";
+import { useRouter } from "System/Router/useRouter";
+
+const TitleStepSchema = Yup.object().shape({
+  title: Yup.string().required().trim(),
+})
+
+const INPUT_SCHEMAS = {
+  "title": TitleStepSchema
+}
+
+interface Actions {
+  goBack: () => void
+  saveAndContinue: () => void
+  submitSubmission: () => void
+  atFirstStep: () => boolean
+  atLastStep: () => boolean
+  currentStep: () => string
+  isSaving: () => boolean
+  isValid: () => boolean
+  onChange: (event: any) => void
+  updatePayload: (input: {}) => void
+  setInitialValues: (values: any) => void
+  isDirty: () => boolean
```

Comment:
re: formik — fair enough! Will circle back with some level of Formik integration.

> Yep, with a route for each step, we could (and probably should) split up the artwork model and use a different model for each step.

Yeah, that's really what I'm after regardless of Formik usage 👍🏼

Reply:
Its cheap to spin up multiple formik contexts per route, and totally fine! Can't count the number of times we've had to unwind custom inputs in favor of formik through the years -- it always gets undone. And by using formik we might be able to use the ArtistAutoComplete component mentioned earlier.

Reply:
> I'm not completely sold on the upsides around consistency / reuse — our inputs are pretty robust even without Formik and I don't think copy/pasting around models/validation between web and app is particularly useful.

I'll try to explain a bit more why I think consistency between the two platforms and re-using code are good ideas.

For me, it's mainly about developer experience and velocity/efficency. In previous projects, I've realized how easy it is to develop a feature in one of these repositories and then just "convert" the code to the other platform (thanks to React/React Native, palette/palett-mobile, formik, routing by URLs, etc.). 

I see two main upsides here. The first is that it makes developing a feature for both platforms, web and app, way faster. The second is that fixing bugs or adding features on top usually involves doing it for the app and web. Keeping things consistent now means that whoever is working on the code doesn't have to understand two different code bases but just one and can use this knowledge to make the changes on both platforms.

Of course, converting the code involves more than just copying and pasting, and it's not always desirable because sometimes both platforms should behave differently, but, well, usually, it's pretty easy.

Reply:
As someone who has been working in a very cross platform way for the past year, the copy/paste-ability between our systems has been astoundingly helpful, and almost like a miracle. So... lets continue doing what we're doing and avoid one-off, custom implementations of things 👍